web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -403,6 +403,15 @@ $.extend(fixmystreet.set_up, {
                 $category_meta.empty();
             }
         });
+
+        if ($(this).data('role') == 'inspector') {
```

Comment:
`fixmystreet.run` is used in set up for moderation etc., that could work here, though it is only one level of checking, but something like that. You need something like `if (fixmystreet.hooks && fixmystreet.hooks.category_change) { fixmystreet.hooks.category_change(args) }`.

Also, just thought - it probably only wants to do the setting if they're currently blank, to prevent removing anything already entered.